exercises/circular-buffer/canonical-data.json

```diff
@@ -0,0 +1,227 @@
+{
+  "#": [
+    "In general, these circular buffers are expected to be stateful,",
+    "and each language will operate on them differently.",
+    "Tests tend to perform a series of operations, some of which expect a certain result.",
+    "As such, this common test suite can only say in abstract terms what should be done.",
+    "",
+    "Tests will contain a number of operations. The operation will be specified in the `operation` key.",
+    "Based on the operation, other keys may be present.",
+    "read: Reading from the buffer should produce the item located at `expected`.",
+    "write: Writing the item located at `item` should succeed.",
+    "readShouldFail: Reading from the buffer should fail.",
+    "writeShouldFail: Writing the item located at `item` should fail.",
+    "overwrite: Write the item located at `item` into the buffer, removing the oldest item if necessary.",
+    "clear: Clear the buffer.",
+    ""
+  ],
+  "cases": [
+    {
+      "description": "reading empty buffer should fail",
+      "capacity": 1,
+      "operations": [
+        {
+          "operation": "readShouldFail"
+        }
+      ]
+    },
+    {
+      "description": "can read an item just written",
+      "capacity": 1,
+      "operations": [
+        {
+          "operation": "write",
+          "item": 1
+        },
+        {
+          "operation": "read",
+          "expected": 1
+        },
+        {
+          "operation": "readShouldFail"
```

Comment:
not related to reading an item just written. A separate test, "Each item can only be read once"?